gradio/components.py

```diff
@@ -2761,6 +2761,138 @@ def style(self, *, full_width: Optional[bool] = None, **kwargs):
         return IOComponent.style(self, **kwargs)
 
 
+@document("change", "style")
+class UploadButton(Changeable, Uploadable, IOComponent, SimpleSerializable):
+    """
+    Used to create an upload button, when cicked allows a user to upload files that satisfy the specified file type or generic files (if file_type not set). The label is the text shown on the button, defaults to "Upload a File".
+    Preprocessing: passes the uploaded file as a {file-object} or {List[file-object]} depending on `file_count` (or a {bytes}/{List{bytes}} depending on `type`)
+    Postprocessing: expects function to return a {str} path to a file, or {List[str]} consisting of paths to files.
+    Examples-format: a {str} path to a local file that populates the component.
+    Demos:
+    """
+
+    def __init__(
+        self,
+        value: Optional[str | List[str] | Callable] = None,
+        *,
+        visible: bool = True,
+        elem_id: Optional[str] = None,
+        type: str = "file",
+        file_type: str = "file",
+        label: str = None,
```

Comment:
Could we have the `UploadButton` support multiple files similar to the `File` component through a `file_count` parameter?

And could we add support for `file_type` in the `File` component? 

Would be good to have a unified interface as much as possible IMO.

Reply:
done